pkg/restore/split.go

```diff
@@ -105,6 +106,9 @@ SplitRegions:
 			var newRegions []*RegionInfo
 			newRegions, err = rs.splitAndScatterRegions(ctx, regionMap[regionID], keys)
 			if err != nil {
+				if strings.Contains(err.Error(), "no valid key") {
```

Comment:
How can it occur? Could you add some comments?

Reply:
The record backup range has appended a "\000" to the end key, TiKV truncates this suffix and BR does not. Will fixed it in this PR.